base/socket.jl

```diff
@@ -2,6 +2,7 @@
 abstract IPAddr
 
 Base.isless{T<:IPAddr}(a::T, b::T) = isless(a.host, b.host)
+Base.convert{T<:Integer}(dt::Type{T}, ip::IPAddr) = dt(ip.host)
```

Comment:
This would more typically be written as 

```jl
Base.convert{T<:Integer}(::Type{T}, ip::IPAddr) = T(ip.host)
```

It doesn't make any functional difference, but reading this initially it took me a bit to figure out what it meant.